app/controllers/certificates_controller.rb

```diff
@@ -0,0 +1,33 @@
+class CertificatesController < ApplicationController
+  include WithCertificateRender
+
+  before_action :authorize_if_private!, only: [:show]
+  before_action :set_certificate!
+  before_action :validate_current_user!, only: [:download]
+
+  helper_method :certificates_current_user?
```

Comment:
This one could be just a certificate method, which I think it would be more clear.